### ISSUE TITLE
[BUG] UNS job is scheduled when unseal copy exists

### ISSUE THREAD

**Issue by benjaminh83** [filecoin-project/lotus#7162]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [lotus forum](https://github.com/filecoin-project/lotus/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to lotus.

### Lotus component

- [ ] lotus daemon - chain sync
- [ ] lotus miner - mining and block production
- [ ] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [ ] lotus miner/market - storage deal
- [X] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### Lotus Version

```text
$ lotus-miner version
Daemon:  1.11.2-dev+mainnet+git.5bdc1862d+api1.2.0
Local: lotus-miner version 1.11.2-dev+mainnet+git.5bdc1862d
```

### Describe the Bug

I see multiple issues from getting a retrieval deal on a 32GiB deal generated by the dealbot. 

 **Issue 1:** 
 My main miner seems to be the only one that can do the "UNS" unseal job. This is very unfortunate, as it takes up a lot of CPU ressource, which could impact windowPoST and winningPoST generation.
 
 I want to put the UNS job on my PC1 worker, but I understand that it needs access to a storage path for "storage" and not only "sealing". Therefore it seems like my PC1 worker would have long term storage, which I do not want.  So what are the options: 
- Put a HDD in my PC1 worker to store unsealed copies?
- Make NFS share to my unsealed storage path on miner and mount it on PC1?
- Maybe let the market node do this in the future and put it on its "cache disk"?

**Issue 2** 
My miner is scheduled with an unseal job, although there is an unsealed copy present in storage. That makes no sense to me. Track of deal: 

- Unseal job scheduled on miner log: 

```
2021-08-23T16:05:09.073+0200	DEBUG	advmgr	sector-storage/manager.go:228	acquire unseal sector lock for sector {2576 5909}
2021-08-23T16:05:09.073+0200	DEBUG	advmgr	sector-storage/manager.go:257	will schedule unseal for sector {2576 5909}
```

- What deal is in that sector: 

```
$ lotus-miner sectors status 5909
SectorID:	5909
Status:		Proving
CIDcommD:	baga6ea4seaqpyq4dfdi3bwkf5iyabxiqx72sg3m7g5zku3pngpqqeljp3r5hehq
CIDcommR:	bagboea4b5abca5hyk76vzx76qhi7x3own2bnfqt3ikvu5kfqcpplpl3w5nncc7bv
Ticket:		e155103336ba067b6158dedf3935da60fefbd5dd5509096d5d536f1ac4889a4e
TicketH:	997438
Seed:		f9a61b7d2c4e323a1b49a73f46a5bf2fe4562372eca0622163a9b51d6b5971a8
SeedH:		998961
Precommit:	bafy2bzaceceyfrkaulgxcfinmj7ap76xudvmre4s5owiqfd7v32u3vucnvujs
Commit:		bafy2bzaceaekdtikzcpljpfxr7pach2imvcvwpnl54yt7lmwraebdx7ycf4mu
Proof:		8ae...2266d
Deals:		[2239065]
Retries:	0
```

- Where is that deal from:

```
$ lotus-miner --call-on-markets storage-deals list -v | grep 2239065
Aug  6 12:02:23  true   bafyreidlwuemhqjwsi72yw73gp67ueeknbcha2txk4vbiw2umfeusoj2qq  2239065  StorageDealActive                        f1wdxdpqh3hirrhp353i4o6ld7bsw6evh3v7i5jtq                                               32GiB     0 FIL                     520455   12D3KooWPfF1DuCGZrt2GKvaxTbW8LB3wMiyHGapKwc5iv7YbxrN-12D3KooWDz6Up4tXg2SVwR54skSV6i4Mh8yBueSx7tRTatuKiihQ-1627600574942283448 
``` 

- Prove that there is a unsealed sector with the name in my long term storage

$ ls /mnt/md122/storage/sealed/s-t02576-5909
/mnt/md122/storage/sealed/s-t02576-5909

$ ls /mnt/md122/storage/unsealed/s-t02576-5909
/mnt/md122/storage/unsealed/s-t02576-5909

### Logging Information

```text
2021-08-23T16:05:09.073+0200	DEBUG	advmgr	sector-storage/manager.go:228	acquire unseal sector lock for sector {2576 5909}
2021-08-23T16:05:09.073+0200	DEBUG	advmgr	sector-storage/manager.go:257	will schedule unseal for sector {2576 5909}
```

### Repo Steps

- Accepted storage deal from Dealbot
- Accepted retrieval deal from Dealbot
- See that it does not pick up the unsealed copy
- See that the unseal job is executed on the main miner, as the PC1 worker is not equipped with long term storage

**Comment by benjaminh83** [comment]
**.lotusminer/sectorstore.json**
```
{
  "ID": "e8417a88-4e43-4d2e-9695-50b6dddac6f7",
  "Weight": 10,
  "CanSeal": false,
  "CanStore": false
}
```
**.lotusminer/storage.json**

```
{
  "StoragePaths": [
    {
      "Path": "/home/acemgt/.lotusminer"
    },
    {
      "Path": "/mnt/md121/storage/"
    },
    {
      "Path": "/mnt/seal/sealing/"
    },
    {
      "Path": "/mnt/md120/storage/"
    },
    {
      "Path": "/mnt/md120/storage2/"
    },
    {
      "Path": "/mnt/md122/storage/"
    }
  ]
}
```

**Comment by benjaminh83** [comment]
Just so we are clear: 

```
ls /mnt/md122/storage/sealed/s-t02576-5909 -lt
-rw-rw-r-- 1 acemgt acemgt 34359738368 Aug  6 21:12 /mnt/md122/storage/sealed/s-t02576-5909

ls /mnt/md122/storage/unsealed/s-t02576-5909 -lt
-rw-r--r-- 1 acemgt acemgt 34359738372 Aug 23 16:05 /mnt/md122/storage/unsealed/s-t02576-5909
```

It will be interesting to see where the UNS job on sector 5909 is going to write the unsealed copy...

**Comment by benjaminh83** [comment]
Here are the last log lines of the UNS (unseal) job that has been running on the main miner (from miner logs). It seem like it has just saved the unsealed copy right where it already was located. Its seems like there must been a glitch, since it did not pickup the file in the first place: 
```
2021-08-23T21:36:51.525 INFO filcrypto::proofs::api > unseal_range: finish
2021-08-23T21:36:51.568+0200	DEBUG	advmgr	sector-storage/worker_local.go:452	worker has unsealed piece, sector={Miner:2576 Number:5909}
2021-08-23T21:36:51.569+0200	DEBUG	advmgr	sector-storage/manager.go:268	completed unseal sector {2576 5909}
2021-08-23T21:36:51.575+0200	INFO	stores	stores/http_handler.go:193	SERVE Alloc check /remote/unsealed/s-t02576-5909/8/allocated/0/34091302912
2021-08-23T21:36:51.575+0200	DEBUG	stores	stores/http_handler.go:287	returning ok: worker has unsealed file with unsealed piece, sector:{Miner:2576 Number:5909}, offset:0, size:34091302912
2021-08-23T21:36:51.575+0200	INFO	stores	stores/http_handler.go:88	SERVE GET /remote/unsealed/s-t02576-5909
2021-08-23T21:37:02.239+0200	DEBUG	advmgr	sector-storage/sched.go:354	SCHED 0 queued; 8 open windows
2021-08-23T21:37:06.051+0200	INFO	miner	miner/miner.go:478	completed mineOne	{"tookMilliseconds": 25, "forRound": 1048035, "baseEpoch": 1048034, "baseDeltaSeconds": 6, "nullRounds": 0, "lateStart": false, "beaconEpoch": 1143879, "lookbackEpochs": 900, "networkPowerAtLookback": "11009184146039701504", "minerPowerAtLookback": "368900430430208", "isEligible": true, "isWinner": false, "error": null}
2021-08-23T21:37:36.164+0200	INFO	miner	miner/miner.go:478	completed mineOne	{"tookMilliseconds": 6, "forRound": 1048036, "baseEpoch": 1048035, "baseDeltaSeconds": 6, "nullRounds": 0, "lateStart": false, "beaconEpoch": 1143880, "lookbackEpochs": 900, "networkPowerAtLookback": "11009184146039701504", "minerPowerAtLookback": "368900430430208", "isEligible": true, "isWinner": false, "error": null}
2021-08-23T21:37:53.470+0200	DEBUG	stores	stores/http_handler.go:161	served sector file/dir, sectorID={Miner:2576 Number:5909}, fileType=unsealed, path=/mnt/md122/storage/unsealed/s-t02576-5909
```

**Comment by benjaminh83** [comment]
@raulk Is this something we can look into and get out of the way? 

I have not seen any other random unseal processes starting up, although there was an unsealed copy in place. So this was probably an odd case or a glitch.

Question still remains - why is the UNS worker required to have a storage path marked for "storage"? Why can't it just use its "seal" path and then hand over the unsealed copy to the miner afterwards? 

Also, I'm having this idea that my "now dedicated server" market node, could actually also hold an UNS worker, so it unseals everything when necessary. This market node also has a 18TiB fast raid SSD for cashing files. I just can't seem to figure out, if there is this opportunity to start collecting unsealed copies at the market node, so we don't need the miner long term storage anymore. 

Having a large array on the market node, starts to feel like this is the future path to a separate retrieval node/miner !?

**Comment by nonsense** [comment]
@benjaminh83 about `Issue 2`: do you have any other information that the miner is indeed unsealing the sector that already has an unsealed copy present in long-term storage?

The two log lines that you pasted are always seen, even if you have an unsealed sector around:

```
acquire unseal sector lock for sector %d
will schedule unseal for sector %d
```

Only after them we check and fetched an unseal copy.

**Comment by benjaminh83** [comment]
@nonsense Sorry no, I don't have anything more I think. But I'm sure it was performing an UNS job, as the miner was basically using resources of a single PC1-job for about 5.5h before the retrieval deal was done. See the time stamps: 
 
```
2021-08-23T16:05:09.073+0200	DEBUG	advmgr	sector-storage/manager.go:228	acquire unseal sector lock for sector {2576 5909}
2021-08-23T16:05:09.073+0200	DEBUG	advmgr	sector-storage/manager.go:257	will schedule unseal for sector {2576 5909}
2021-08-23T21:36:51.525 INFO filcrypto::proofs::api > unseal_range: finish
2021-08-23T21:36:51.568+0200	DEBUG	advmgr	sector-storage/worker_local.go:452	worker has unsealed piece, sector={Miner:2576 Number:5909}
2021-08-23T21:36:51.569+0200	DEBUG	advmgr	sector-storage/manager.go:268	completed unseal sector {2576 5909}
```

**Comment by jennijuju** [comment]
Are you still seeing this issue? if so - please provide reproduce steps for trouble shooting.

**Comment by benjaminh83** [comment]
I have not seen my miner fail to acquire an unsealed sector and triggering a UNS job since this event. 
I do remember @stuberman describing pretty specifically how an UNS job is failing on a Worker unless it has a mounted can-store path. I can't find it, so I suggest we do some ned UNS testing and narrow down the actual issue from using sealing-workers to do UNS jobs.

**Comment by benjaminh83** [comment]
I have not experienced issue 2 since, so I would not consider it a bug.

Issue 1 has been re-addressed in this issue #7763 so we can close this one and continue in there.
